Remove duplicate ADPs. Solving issue #240

perezp opened the pull request:
This PR avoids returning replicated ADPs in case that there exist just a very few cloud combinations. 

It is intended to solve issue #240 

It includes tests for the AAM that raised the issue; an AAM with only one module and just one cloud offer as possibility.

andreaturli commented on planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/OptimizerInitialDeployment.java:
```diff
@@ -108,9 +108,9 @@ private String openBenchmarkOffersDefaultFile() {
-      log.debug("AAM is: " + appModel);
-      log.debug("Suitable offers are: " + suitableCloudOffer);
-      log.debug("Informaton of Benchmark platform is: " + benchmarkPlatformsYaml);
+      log.debug("AAM is: {}", appModel);
```
- try to avoid multi-lines logs. 
- `log.info` is more [appropriate](http://stackoverflow.com/questions/6860186/log-info-vs-log-debug)

andreaturli commented on planner/optimizer/optimizer-core/src/main/java/eu/seaclouds/platform/planner/optimizer/heuristics/AbstractHeuristic.java:
```diff
@@ -34,9 +34,9 @@
-   private int MAX_ITER_NO_IMPROVE = 200; // 200;
-   private double MAX_TIMES_IMPROVE_REQUIREMENT = 20;
-   private static final int DEFAULT_MAX_NUM_INSTANCES = 10;
+   private int              MAX_ITER_NO_IMPROVE           = 200; // 200;
```
weird indentation

perezp commented:
Thank you @andreaturli . 
I've addressed both comments and updated the PR (although I'll take care of the indentation settings in the IDE, the issue about the indentation in attributes definition is prone to be repeated in the future).

andreaturli commented:
lgtm
